packages/react/src/components/Forms/DatePicker/Calendar.tsx

```diff
@@ -0,0 +1,323 @@
+import React, { useEffect, useMemo, useRef } from "react";
+import {
+  CalendarDate,
+  createCalendar,
+  DateValue,
+  GregorianCalendar,
+  toCalendar,
+} from "@internationalized/date";
+import { useDateFormatter, useLocale } from "@react-aria/i18n";
+import { useCalendarState } from "@react-stately/calendar";
+import { useCalendar } from "@react-aria/calendar";
+import {
+  useSelect,
+  UseSelectReturnValue,
+  UseSelectStateChange,
+} from "downshift";
+import classNames from "classnames";
+import { CalendarProps } from "react-aria";
+import { range } from ":/utils";
+import { Button } from ":/components/Button";
+import { CalendarGrid } from ":/components/Forms/DatePicker/CalendarGrid";
+import { useCunningham } from ":/components/Provider";
+
+// todo to be factorized with the select component
+interface Option {
+  value: number;
+  label: string;
+  disabled?: boolean;
+}
+
+// todo to be factorized with the select component
+const optionToString = (option: Option | null) => {
+  return option ? option.label : "";
+};
+
+type DropdownValuesProps = {
+  options: Array<Option>;
+  downShift: UseSelectReturnValue<Option>;
+};
+
+// todo to be factorized with the select component
+const DropdownValues = ({ options, downShift }: DropdownValuesProps) => (
+  <div
+    className={classNames("c__calendar__menu", {
+      "c__calendar__menu--opened": downShift.isOpen,
+    })}
+    {...downShift.getMenuProps()}
+  >
+    <ul>
+      {downShift.isOpen &&
+        options.map((item, index) => (
+          <li
+            key={`${item.value}${index}`}
+            {...downShift.getItemProps({
+              item,
+              index,
+              disabled: item.disabled,
+            })}
+            className={classNames("c__calendar__menu__item", {
+              "c__calendar__menu__item--highlight":
+                downShift.highlightedIndex === index,
+              "c__calendar__menu__item--selected":
+                downShift.selectedItem?.label === item.label,
+              "c__calendar__menu__item--disabled": item.disabled,
+            })}
+          >
+            <span>{item.label}</span>
+          </li>
+        ))}
+    </ul>
+  </div>
+);
+
+interface CalendarSubProps extends CalendarProps<DateValue> {
+  minYear?: number;
+  maxYear?: number;
+}
+
+export const Calendar = ({
+  minYear = 1800, // in gregorian calendar.
```

Comment:
I realized that `1900` is enough I think
```suggestion
  minYear = 1900, // in gregorian calendar.
```